Page layouts and routing.

samyukthasudhakar opened the pull request:
The PR has the following screens :
- Home
- Playlist
- Liked
- Watch Later
- History

Routes have been set to these pages.

Note : Functionalities have not been enabled for any of the pages yet. This is just the skeleton.

Files to review are part of this [commit](https://github.com/samyukthasudhakar/video-library/commit/89876667a0170a6f2a91d4ceed551654467ce025) .

Styles for all the components and pages are part of this [commit](https://github.com/samyukthasudhakar/video-library/commit/b822f896888450d5d974cfc551b6739b427d4511) . Feel free to skip the styles as they are mostly reused from CL and ecommerce.

Live : https://video-library-pages.netlify.app/

samyukthasudhakar committed:
refactor : cleaned default styles and pages

samyukthasudhakar committed:
assets : images and icons added

samyukthasudhakar committed:
styles : added styles to all the components and pages

samyukthasudhakar committed:
feat : created layout for all pages and established routes to these pages

samyukthasudhakar committed:
fix : to make deployment error free

vrinda-mahajan commented on src/components/menuPopUp/menuPopUp.jsx:
```diff
@@ -0,0 +1,21 @@
+import {
+    playlist,
+    watchLater,
+} from "../../assets/icons";
```
suggestion:- use absolute path instead of relative as it increases the code readability.

samyukthasudhakar replied:
Have made changes to use absolute paths.

vrinda-mahajan commented on src/components/sideBar/sideBar.jsx:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <aside className="side-navigation">
+      <ul>
+        {navLinks.map((navLink) => {
+          return (
+            <Link to={navLink.link}
+              className="flex-column flex-center nav-pill"
+            >
+              <li>
+                <img className="nav-img" src={navLink.icon} />
+              </li>
+              <li className="nav-text">
+                {navLink.name}
```
Suggestion: you can use destructuring here in map `navLinks.map(({name, icon, link}) =>`

samyukthasudhakar replied:
Used destructuring as suggested.

vrinda-mahajan commented on src/pages/history/history.jsx:
```diff
@@ -0,0 +1,18 @@
+            <Header />
+            <div className="flex-layout">
+                <SideNavBar /> 
+                <div style={{backgroundColor:"#f9f9f9", width:"100vw"}} >
```
try to avoid inline styling

samyukthasudhakar replied:
Removed all inline styling used.

vrinda-mahajan commented on src/pages/home/home.css:
```diff
@@ -0,0 +1,15 @@
+    padding: 7px;
+}
+
+/* #f9f9f9 */
```
remove this comment.

vrinda-mahajan commented on src/pages/playlist/playlist.jsx:
```diff
@@ -0,0 +1,22 @@
+        <div>
+            <Header />
+            <div className="flex-layout">
+                <SideNavBar /> 
```
suggestion: if the header and sidebar are present on every page put them into the `app.js` file.

samyukthasudhakar replied:
Sure, done.

vrinda-mahajan reviewed (approved):
The code is clean and readable. UI is also looking great. Just added a few suggestions.

vrx29 commented on src/components/sideBar/sideBar.jsx:
```diff
@@ -0,0 +1,41 @@
+      <ul>
+        {navLinks.map((navLink) => {
+          return (
+            <Link to={navLink.link}
```
please provide key whenever you are using map function. If you dont have unique id property you can use. index as second parameter of map function as key

samyukthasudhakar replied:
Have made use of index parameter as suggested.

vrx29 commented on src/components/videoCard/videoCard.jsx:
```diff
@@ -0,0 +1,34 @@
+import {menu} from '../../assets/icons'
+import './videoCard.css'
+
+import { MenuPopUp } from '../menuPopUp/menuPopUp';
```
instead of creating same component as folder name you can use index.js in menuPopUp folder it will also work properly and while importing you have to write menuPopUp only once

samyukthasudhakar replied:
Yes, this is much better. Thanks for pointing it out, have made changes accordingly.

samyukthasudhakar committed:
• changed relative paths to absolute
• implemented destructuring and passing of key while using map on nav links
• removed inline styling used
• removed comment
• changed component and page jsx files within folders to index.jsx for easier imports
